### PR TITLE
eslint: add typescript consistent-type-assertions rule

### PR DESCRIPTION
**Pull request by quisquous**

### CODE REVIEW

File: resources/party.ts [hunk 1]
```diff
@@ -10,7 +19,7 @@ export default class PartyTracker {
   allianceIds_: string[] = [];
   nameToRole_: Record<string, Role> = {};
   idToName_: Record<string, string> = {};
-  roleToPartyNames_: Record<Role, string[]> = {} as Record<Role, string[]>;
+  roleToPartyNames_: Record<Role, string[]> = emptyRoleToPartyNames;
```
Comment (Trim21) [inline resources/party.ts:22]: Won't this object be shared cross `PartyTracker` instance?

Comment (quisquous) [reply resources/party.ts]: Oh whoops, thanks!

Comment (Trim21) [reply resources/party.ts]: I'm not sure if it works after your changes...

Copy and original object share array, too.

Comment (quisquous) [reply resources/party.ts]: Thanks for catching my (repeated) mistakes.  *<_<*

**Review by panicstevenson — APPROVED** [review]

**Review by Trim21 — APPROVED** [review]

**Pull request merged**